evalml/tests/component_tests/test_lgbm_regressor.py

```diff
@@ -0,0 +1,225 @@
+    clf.fit(X, y)
+    clf = LightGBMRegressor(n_estimators=1, max_depth=1, random_state=SEED_BOUNDS.max_bound)
+    clf.fit(X, y)
+
+
+def test_lightgbm_regressor_random_state_bounds_rng(X_y_regression):
+    """when a RNG is inputted for random_state, ensure the sample we take to get a random seed for lightgbm is in lightgbm's supported range"""
+
+    def make_mock_random_state(return_value):
+
+        class MockRandomState(np.random.RandomState):
+
+            def randint(self, min_bound, max_bound):
+                return return_value
+        return MockRandomState()
+
+    X, y = X_y_regression
+    col_names = ["col_{}".format(i) for i in range(len(X[0]))]
+    X = pd.DataFrame(X, columns=col_names)
+    y = pd.Series(y)
+    rng = make_mock_random_state(LightGBMRegressor.SEED_MIN)
+    clf = LightGBMRegressor(n_estimators=1, max_depth=1, random_state=rng)
+    clf.fit(X, y)
+    rng = make_mock_random_state(LightGBMRegressor.SEED_MAX)
+    clf = LightGBMRegressor(n_estimators=1, max_depth=1, random_state=rng)
+    clf.fit(X, y)
+
+
+def test_fit_predict_regression(X_y_regression):
+    X, y = X_y_regression
+
+    sk_clf = lgbm.sklearn.LGBMRegressor(random_state=0)
+    sk_clf.fit(X, y)
+    y_pred_sk = sk_clf.predict(X)
+
+    clf = LightGBMRegressor()
+    clf.fit(X, y)
+    y_pred = clf.predict(X)
+
+    np.testing.assert_almost_equal(y_pred, y_pred_sk, decimal=3)
+
+
+def test_feature_importance(X_y_regression):
+    X, y = X_y_regression
+
+    clf = LightGBMRegressor()
+    sk_clf = lgbm.sklearn.LGBMRegressor(random_state=0)
+    sk_clf.fit(X, y)
+    sk_feature_importance = sk_clf.feature_importances_
+
+    clf.fit(X, y)
+    feature_importance = clf.feature_importance
+
+    np.testing.assert_almost_equal(sk_feature_importance, feature_importance, decimal=3)
+
+
+def test_fit_string_features(X_y_regression):
+    X, y = X_y_regression
+    X = pd.DataFrame(X)
+    X['string_col'] = 'abc'
+
+    # lightGBM requires input args to be int, float, or bool, not string
+    X_expected = X.copy()
+    X_expected['string_col'] = 0.0
+
+    clf = lgbm.sklearn.LGBMRegressor(random_state=0)
+    clf.fit(X_expected, y, categorical_feature=['string_col'])
+    y_pred_sk = clf.predict(X_expected)
+
+    clf = LightGBMRegressor()
+    clf.fit(X, y)
+    y_pred = clf.predict(X)
+
+    np.testing.assert_almost_equal(y_pred, y_pred_sk, decimal=3)
+
+
+@patch('evalml.pipelines.components.estimators.estimator.Estimator.predict')
+@patch('evalml.pipelines.components.component_base.ComponentBase.fit')
+def test_fit_no_categories(mock_fit, mock_predict, X_y_regression):
+    X, y = X_y_regression
+    X2 = pd.DataFrame(X)
+    X2.columns = np.arange(len(X2.columns))
+    clf = LightGBMRegressor()
+    clf.fit(X, y)
+    arg_X = mock_fit.call_args[0][0]
+    np.testing.assert_array_equal(arg_X, X2)
+
+    clf.predict(X)
+    arg_X = mock_predict.call_args[0][0]
+    np.testing.assert_array_equal(arg_X, X2)
```

Comment:
@bchen1116 could you remind me what this unit test is for?

Reply:
Mm good call, I had originally put this in as a test for no categorical data, but a few tests prior to this covers that instance. I'll fix this!